cdk/backend/function/expressLambda/api/questionAnswers/questionAnswersQuery.ts

```diff
@@ -0,0 +1,281 @@
+
+  return {
+    message: `🚀 ~ > QuestionAnswer with id: ${id}`,
+    data: response,
+  }
+}
+
+const createQuestionAnswer = async ({
+  userFormId,
+  questionId,
+  knowledge,
+  motivation,
+  customScaleValue,
+  textValue,
+}: CreateQuestionAnswerProp) => {
+  const id = uuidv4()
+
+  const params: SqlParameter[] = [
+    {
+      name: 'id',
+      value: {
+        stringValue: id,
+      },
+      typeHint: TypeHint.UUID,
+    },
+    {
+      name: 'userFormId',
+      value: {
+        stringValue: userFormId,
+      },
+      typeHint: TypeHint.UUID,
+    },
+    {
+      name: 'questionId',
+      value: {
+        stringValue: questionId,
+      },
+      typeHint: TypeHint.UUID,
+    },
+    {
+      name: 'knowledge',
+      value: {
+        longValue: knowledge,
+      },
+    },
+    {
+      name: 'motivation',
+      value: {
+        longValue: motivation,
+      },
+    },
+    {
+      name: 'customScaleValue',
+      value: customScaleValue
+        ? {
+            longValue: customScaleValue,
+          }
+        : { isNull: true },
+    },
+    {
+      name: 'textValue',
+      value: textValue ? { stringValue: textValue } : { isNull: true },
+    },
+  ]
+
+  const query = `INSERT INTO questionAnswer (id, userFormID, questionID, knowledge, motivation, customScaleValue)
+    VALUES(:id, :userFormId, :questionId, :knowledge, :motivation, :customScaleValue)
+    RETURNING *`
+  const response = await sqlQuery(query, params)
+
+  return {
+    message: `🚀 ~ > QuestionAnswer with id: ${id} was successfully created`,
+    data: response,
+  }
+}
+
+const updateQuestionAnswer = async (
+  id: string,
+  {
+    userFormId,
+    questionId,
+    knowledge,
+    motivation,
+    customScaleValue,
+    textValue,
+  }: CreateQuestionAnswerProp
+) => {
+  const params: SqlParameter[] = [
+    {
+      name: 'id',
+      value: {
+        stringValue: id,
+      },
+      typeHint: TypeHint.UUID,
+    },
+    {
+      name: 'userFormId',
+      value: {
+        stringValue: userFormId,
+      },
+      typeHint: TypeHint.UUID,
+    },
+    {
+      name: 'questionId',
+      value: {
+        stringValue: questionId,
+      },
+      typeHint: TypeHint.UUID,
+    },
+    {
+      name: 'knowledge',
+      value: {
+        longValue: knowledge,
+      },
+    },
+    {
+      name: 'motivation',
+      value: {
+        longValue: motivation,
+      },
+    },
+    {
+      name: 'customScaleValue',
+      value: customScaleValue
+        ? {
+            longValue: customScaleValue,
+          }
+        : { isNull: true },
+    },
+    {
+      name: 'textValue',
+      value: textValue ? { stringValue: textValue } : { isNull: true },
+    },
+  ]
+
+  const query = `UPDATE questionAnswer
+        SET userFormID=:userFormId, questionID=:questionId, knowledge=:knowledge, motivation=:motivation,
+        customScaleValue=:customScaleValue, textValue=:textValue
+        WHERE id=:id
+        RETURNING *`
```

Comment:
Arrester meg om jeg tar feil, men om jeg forstår update-spørringen riktig så:

1. For at du skal oppdatere ett enkelt felt her, så må du likevel sende inn alle de andre sine gamle verdier (i og med at de er definert i strengen.
2. Du må da først spørre etter all dataen, deretter sende inn den gamle dataen pluss det nye feltet.

Det er trolig litt overkill, men det er ikke så veldig enkelt å lage et clean dynamisk endepunkt her uten en query-builder. Vi kan ikke ta inn direkte fra brukeren hvilke felter og generere en query-streng heller, for det er supersårbart for SQL-injections.

Om du ser her så kan du se hvordan jeg løste det i `formDefinition`. Tror jeg skal skrive en mer gjenbrukbar variant av denne slik at man kan bruke den her og i alle andre liknende tilfeller også.
https://github.com/knowit/kompetansekartlegging-app/blob/feat/express-lambda/cdk/backend/function/expressLambda/api/formDefinition/formDefinitionQueries.ts#L85

Bare å ha i bakhodet iaf, 

lgtm 🎉

Reply:
Ja mulig det er bedre å gjøre det på den måten. Det jeg tenkte var at når man oppdaterer et `questionAnswer`så har man allerede de gamle verdiene i frontend så man kan sende de tilbake sammen med den/de oppdaterte, men mulig det blir mye rot i frontend.